src/ucp/wireup/address.h

```diff
@@ -76,6 +76,7 @@ struct ucp_address_entry {
     uint16_t                    tl_name_csum;   /* Checksum of transport name */
     ucp_rsc_index_t             md_index;       /* Memory domain index */
     ucp_rsc_index_t             dev_index;      /* Device index */
+    unsigned                    dev_concurrency;    /* Device concurrency count */
```

Comment:
So, is the intend to open multiple EP or Workers ? EP will be sterilized on worker, right ? You really need multiple threads to drive PMIO to benefit from this, which means you need multiple workers. BTW, it can be very beneficial for socket transport.

Reply:
we want to open multiple eps for same pair of local,remote worker
what is PMIO?
yes, it would probably help socket transport

Reply:
PMIO - inline doorbell triggered through cpu data copy (no DMA).  I'm still concerned how it will work when you have sterilization in other places ? Why multiple worker approach is not enough ? Do we really need multiple endpoints or the endpoint just can represent multiple QPs.  It would be helpful to understand details of the problem  so we can do educated engineering decision. I'm not sure I have enough details at the moment.